docs/proposals/image-promotion.md

```diff
@@ -0,0 +1,152 @@
+# Image Promotion
+
+## Abstract
+
+This proposal describes the best practices and defines patterns for promoting Docker
+images between different stages (eg. from "development" to "production").
+This proposal also describes priciples a way how to configure OpenShift to serve as
+a Continuous Delivery tool.
+
+## Constraints and Assumptions
+
+This document describes the following ways to promote images
+
+* Based on human intervention
+* Driven purely by OpenShift
+* Based on external Continuous Delivery tool (Jenkins, go.cd, Snap, ...)
```

Comment:
you describe these in a different order than you list them here, which is confusing.

i'd describe them in this order:
1) based on human intervention (using all the existing oc and docker commands that exist today, manually)
2) automated by openshift mechanisms (using deployment hooks, possibly with some new functionality) - i don't think you really elaborate on this method below.
3) using an external tool

also i found the "semi automatic" scenario described below a bit confused...it didn't really map to any of the 3 methods listed here.  Aside from the fact that i know you're trying to ensure *our* use case is covered, i'd be in favor of dropping it or making it fit into one of the three mechanisms listed here.  it's not clear to me why it's unique.